src/rez/package_cache.py

```diff
@@ -214,17 +230,43 @@ def add_variant(self, variant, force=False):
                     % package.repository
                 )
 
-        no_op_statuses = (
+        no_op_statuses = {
             self.VARIANT_FOUND,
-            self.VARIANT_COPYING,
-            self.VARIANT_COPY_STALLED
-        )
+            self.VARIANT_COPY_STALLED,
+        }
+        if not wait_for_copying:
+            # Copying variants are only no-ops if we want to ignore them.
+            no_op_statuses.add(self.VARIANT_COPYING)
 
         # variant already exists, or is being copied to cache by another thread/proc
         status, rootpath = self._get_cached_root(variant)
         if status in no_op_statuses:
+            if logger:
+                logger.warning(f"Not caching {variant.qualified_name}. "
+                               f"Variant {self.STATUS_DESCRIPTIONS[status]}")
             return (rootpath, status)
 
+        if wait_for_copying and status == self.VARIANT_COPYING:
+            spinner = PixelSpinner(f"Waiting for {variant.qualified_name} to finish copying. ")
```

Comment:
If I'm not mistaken, the `progress` package takes care of disabling spinners if there is no TTY attached the the process. And I also think the output goes to stderr by default. Could you confirm both please? I don't want this code to spam logs in CI or in a render farm and I also don't want to see the spinner in stdout since that would affect the pipebility (if that's a word...) of rez.

Reply:
Yeah, that appears to be the case.